### ISSUE TITLE
Potential OutOfMemory

### ISSUE THREAD

**Issue by dkomanov** [jasync-sql/jasync-sql#187]
Exception is thrown here: https://github.com/jasync-sql/jasync-sql/blob/master/mysql-async/src/main/java/com/github/jasync/sql/db/mysql/codec/MySQLConnectionHandler.kt#L153. If it happens, allocated buffer won't be freed.

It actually happened, but I don't know root cause yet, in any case it's better to fail more gracefully.

**Comment by oshai** [comment]
This looks like some kind of misuse of connection or a bug. What buffer do you see does not cleaned up? I am not sure it's easy to clean as the connection state is corrupted.

**Comment by dkomanov** [comment]
I think release of resources and closing of a connection with proper error propagation is much better then resource leak.

In any case, I don't know yet, what caused it.

**Comment by dkomanov** [comment]
My guess that system was under load and we already had problems with memory, so corrupted connection could be caused by another OutOfMemoryError and this code just makes it slightly worse.

**Comment by oshai** [comment]
If I compare it to JDBC for example, it's the responsibility of the user to close the connection gracefully in case of an error.

**Comment by dkomanov** [comment]
Right, but in this case my code can't do anything with leaked memory. Even if I close connection eventually.

**Comment by oshai** [comment]
Do you want to suggest a PR to fix? I am not sure I understand what you mean.

**Comment by oshai** [comment]
@dkomanov any update on the issue?

**Comment by dkomanov** [comment]
I don't have time for it now. I'll be back to it eventually.
